python/lsst/summit/utils/guiders/reading.py

```diff
@@ -0,0 +1,724 @@
+        print(f"ROI Shape (row, col): {header['roi_rows']}, {header['roi_cols']}")
+        mystr = f"With nstamps {int(header['n_stamps'])}"
+        mystr = mystr + f" at {header['FREQ']} Hz"
+        print(mystr)
+        print(f"Acq. Start Time: {header['start_time']}")
+
+    @property
+    def guiderNames(self):
+        """Get the names of the guider detectors."""
+        return list(self.guiderNameMap.keys())
+
+    @property
+    def guiderIds(self):
+        """Get the ids of the guider detectors."""
+        return list(self.guiderNameMap.values())
+
+
+# TODO: Check missing stamps
+def getGuiderStamps(
+    detNum: int,
+    seqNum: int,
+    dayObs: int,
+    butler: Butler,
+    view: str = "dvcs",
+    nstamps: int | None = None,
+    whichstamps: list[int] | None = None,
+) -> Stamps:
+    """
+    This class reads the stamp object from the Butler for one Guider and
+    converts them to DVCS view, making a new Stamps object
+
+    Parameters
+    ----------
+    detNum : `int`
+        Detector Id
+    seqNum : `int`
+        Sequence Number
+    dayObs : `int`
+        Day Observation
+    repo : `str`
+        Butler repo
+    collections : `list` of `str`
+        Butler collections
+    butler : `lsst.daf.butler.Butler`, optional
+        Butler object. If None, a new Butler will be created.
+    view : `str`, optional
+        View type, either 'dvcs' or 'ccd'. Default is 'dvcs'.
+    whichstamps : `list` of `int`, optional
+        List of stamp indices to read. If None, all stamps will be read.
+
+    Returns
+    -------
+    stamps : `lsst.meas.algorithms.stamps.Stamps`
+        Stamp images oriented in DVCS or CCD view, depending on the `view`
+        parameter.
+    """
+    # get Camera object
+    camera = LsstCam.getCamera()
+    detector = camera[detNum]
+
+    # for dayObs of 20250509 or before, the ROIs are swapped between SG0 and
+    # SG1. Fix here
+    ccd_swapped: str = ""
+    if dayObs < 20250509:
+        detName = camera[detNum].getName()
+        raft = detName[0:3]
+        ccd = detName[4:7]
+        if ccd == "SG0":
+            ccd_swapped = "SG1"
+        elif ccd == "SG1":
+            ccd_swapped = "SG0"
+
+        detName_swapped = raft + "_" + ccd_swapped
+        detector_swapped = camera[detName_swapped]
+        detNum_swapped = detector_swapped.getId()
+
+        dataId = {
+            "instrument": "LSSTCam",
+            "detector": detNum_swapped,
+            "day_obs": dayObs,
+            "seq_num": seqNum,
+        }
+    else:
+        dataId = {
+            "instrument": "LSSTCam",
+            "detector": detNum,
+            "day_obs": dayObs,
+            "seq_num": seqNum,
+        }
+
+    # finally read from the Butler
+    try:
+        raw_stamps = butler.get("guider_raw", dataId)
+    except DatasetNotFoundError:
+        raise StampsNotFoundError(
+            f"No guider data found for dayObs {dayObs}, seqNum {seqNum}, detector {detNum}."
+        )
```

Comment:
Change `except DatasetNotFoundError` to `except DatasetNotFoundError as e` and then `raise from e` to add your error message to the original one.

Reply:
Got it.